deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/HostTerminator.java

```diff
@@ -66,8 +70,19 @@ private void terminateHost(HostBean host) throws Exception {
 
         if (stopSucceeded) {
             LOG.info(String.format("Host %s is stopped. Terminate it.", hostId));
-            String clusterName = host.getGroup_name();
+
             Boolean replaceHost = host.getState() == HostState.PENDING_TERMINATE;
+            String clusterName = host.getGroup_name();
```

Comment:
why do we still need this line?

Reply:
Also, to cover this change, can you please include terminating host outside of Teletraan(ex: terminate using aws cli) in the test plan?

Reply:
`hosts_and_agents` table doesn't have the entry until the agent pings Teletraan.